proxy/envoy/config.go

```diff
@@ -158,62 +162,88 @@ func buildSidecar(env proxy.Environment, sidecar proxy.Node) (Listeners, Cluster
 	services := env.Services()
 	managementPorts := env.ManagementPorts(sidecar.IPAddress)
 
-	inbound, inClusters := buildInboundListeners(env.Mesh, sidecar, instances)
-	outbound, outClusters := buildOutboundListeners(env.Mesh, sidecar, instances, services, env)
-	mgmtListeners, mgmtClusters := buildMgmtPortListeners(env.Mesh, managementPorts, sidecar.IPAddress)
-
-	listeners := append(inbound, outbound...)
-	clusters := append(inClusters, outClusters...)
-
-	// If management listener port and service port are same, bad things happen
-	// when running in kubernetes, as the probes stop responding. So, append
-	// non overlapping listeners only.
-	for i := range mgmtListeners {
-		m := mgmtListeners[i]
-		c := mgmtClusters[i]
-		l := listeners.GetByAddress(m.Address)
-		if l != nil {
-			glog.Warningf("Omitting listener for management address %s (%s) due to collision with service listener %s (%s)",
-				m.Name, m.Address, l.Name, l.Address)
-			continue
+	listeners := make(Listeners, 0)
+	clusters := make(Clusters, 0)
+
+	if env.Mesh.ProxyListenPort > 0 {
```

Comment:
As I mentioned, I don't think the 'proxy listener' should be optional feature. 
Even if it is - we should be able to generate a RDS for "all", for raw VMs that use manual configuration.